### PR TITLE
Azure rm version upgrade

### PR DESCRIPTION
**Pull request by AnilDhawaleGitHub**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: asp.tf [hunk 1]
```diff
@@ -3,12 +3,12 @@ locals {
 }
 /*module "appServicePlan" {
   source = "git@github.com:hmcts/cnp-module-app-service-plan?ref=master"
-  location = "${var.location}"
-  env = "${var.env}"
+  location = var.location
```
Comment (amitchidrewar1301) [inline asp.tf]: No need for this changes

File: application-insights.tf [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+  name                = join("-", [var.product, var.env])
+  location            = var.appinsights_location
+  resource_group_name = azurerm_resource_group.rg.name
+  application_type    = var.application_type
 }
 
 output "appInsightsInstrumentationKey" {
   value = "${azurerm_application_insights.appinsights.instrumentation_key}"
-}
+}
```
Comment (amitchidrewar1301) [inline application-insights.tf]: Remove newline at the end, pls

File: key-vault.tf [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
-  managed_identity_object_id = "${var.managed_identity_object_id}"
+  common_tags         = var.common_tags
+  managed_identity_object_ids = ["${var.managed_identity_object_id}"]
 
 }
 
 output "vaultName" {
   value = "${module.bar-vault.key_vault_name}"
 }
+
```
Comment (amitchidrewar1301) [inline key-vault.tf]: Remove newline at the end, pls

File: state.tf [hunk 4]
```diff
@@ -8,9 +8,9 @@ data "terraform_remote_state" "core_apps_infrastructure" {
   backend = "azurerm"
 
   config {
-    resource_group_name  = "mgmt-state-store-${var.subscription}"
-    storage_account_name = "mgmtstatestore${var.subscription}"
-    container_name       = "mgmtstatestorecontainer${var.env}"
+    resource_group_name  = join("-",["mgmt-state-store",var.subscription])
+    storage_account_name = join("",["mgmtstatestore",var.subscription])
```
Comment (amitchidrewar1301) [inline state.tf]: better to remove join changes from line 12,13 pls, as it's not having any concatenation char like '-'

File: application-insights.tf [hunk 5]
```diff
@@ -1,10 +1,10 @@
 resource "azurerm_application_insights" "appinsights" {
```
Comment (amitchidrewar1301) [inline application-insights.tf:1]: also please check why checks are failing as well.

File: asp.tf (hunk 1 above)
Comment (AnilDhawaleGitHub) [reply asp.tf]: its done

File: application-insights.tf (hunk 2 above)
Comment (AnilDhawaleGitHub) [reply application-insights.tf]: removed

File: key-vault.tf (hunk 3 above)
Comment (AnilDhawaleGitHub) [reply key-vault.tf]: removed

File: state.tf (hunk 4 above)
Comment (AnilDhawaleGitHub) [reply state.tf]: removed

File: main.tf [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
-    "Destroy Me" = "${var.destroy_me}"
+  tags = {
+    "Deployment Environment" = var.env
+    "Team Name" = var.team_name
+    "Team Contact" = var.team_contact
+    "Destroy Me" = var.destroy_me
     "fake1" = "fake1"
   }
-}
+}
```
Comment (amitchidrewar1301) [inline main.tf]: please remove the new line on main.tf

Comment (AnilDhawaleGitHub) [reply main.tf]: Removed

File: key-vault.tf [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
-  source = "git@github.com:hmcts/cnp-module-key-vault?ref=master"
-  name = "${var.product}-${var.env}"
-  product = "${var.product}"
-  env = "${var.env}"
-  tenant_id = "${var.tenant_id}"
-  object_id = "${var.jenkins_AAD_objectId}"
-  resource_group_name = "${azurerm_resource_group.rg.name}"
+  source = "git@github.com:hmcts/cnp-module-key-vault?ref=azurermv2"
+  name = join("-", [var.product, var.env])
+  product = var.product
```
Comment (vijayrajagopalan-hmcts) [inline key-vault.tf:4]: Replace var.product with **join("-", [var.product, "shared"])**

Comment (AnilDhawaleGitHub) [reply key-vault.tf:4]: we are using the same rg name

**Review by vijayrajagopalan-hmcts — APPROVED** [review]

**Review by amitchidrewar1301 — APPROVED** [review]

**Review by avi-hmcts — APPROVED** [review]

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**